includes/class-newspack-popups-newsletters.php

```diff
@@ -77,6 +99,55 @@ public static function handle_newsletter_subscription( $provider, $contact, $lis
 			$api->save_reader_events( $client_id, [ $subscription_event ] );
 		}
 	}
+
+	/**
+	 * When a reader provides an email address through the Newspack auth flow and
+	 * the reader's newsletter subscription status is unknown, check the status
+	 * of the email address with the ESP currently active in Newspack Newsletters.
+	 *
+	 * @param string|null $email_address Email address or null if not set.
+	 */
+	public static function check_reader_newsletter_subscription_status( $email_address ) {
+		if ( self::$is_checking_status ) {
+			return;
+		}
+		self::$is_checking_status = true;
+		if ( $email_address && class_exists( '\Newspack_Newsletters' ) && class_exists( '\Newspack_Newsletters_Subscription' ) ) {
```

Comment:
We should exit early here, following the standard practice already present in this method.

Reply:
Done in e711c15f13c90cd52d4a992f79c4b42369d03790